src/inputs/ObjectSearch.tsx

```diff
@@ -233,7 +234,9 @@ class ObjectSearch extends Component<IObjectSearchProps> {
     // Select from search
     if (this.isMultiSelect) {
       const selectedOptionIds = selectedOption.map((_selectedOption: any) => _selectedOption.key)
-        , foundOptions = this.options.filter(option => selectedOptionIds.includes(option.id));
+        , optionsToSearch = Array.from(new Set([...this.previousSelectedOptions, ...this.options]))
+        , foundOptions = optionsToSearch.filter((option: any) => selectedOptionIds.includes(option.id));
+      this.previousSelectedOptions = foundOptions;
```

Comment:
Would this break on a non-empty form? Aren't `previousSelectedOptions ` in `value`?

Reply:
Yeah, that's a good catch, will update.